bin/SleepRayTune.py

```diff
@@ -0,0 +1,119 @@
+import shutil
+import hydra
+from omegaconf import DictConfig
+
+import torch
+import logging
+import os
+import pytorch_lightning as pl
+from pytorch_lightning.loggers import TensorBoardLogger
+
+from ray import tune
+from ray.tune import CLIReporter
+from ray.tune.schedulers import ASHAScheduler
+from ray.tune.integration.pytorch_lightning import TuneReportCallback
+
+from bliss import sleep
+from bliss.datasets.simulated import SimulatedModule
+
+
+# define the callback that monitors val_loss
+callback = TuneReportCallback({"loss": "val_loss"}, on="validation_end")
+
+
+def SleepRayTune(config, cfg: DictConfig, num_epochs, num_gpu):
+    # set up the config for SleepPhase
+    cfg.model.encoder.params["enc_conv_c"] = config["enc_conv_c"]
+    cfg.model.encoder.params["enc_kern"] = config["enc_kern"]
+    cfg.model.encoder.params["enc_hidden"] = config["enc_hidden"]
+    cfg.optimizer.params["lr"] = config["lr"]
+    cfg.optimizer.params["weight_decay"] = config["weight_decay"]
+
+    # model
+    model = sleep.SleepPhase(cfg)
+
+    # data module
+    dataset = SimulatedModule(cfg=cfg)
+
+    # parameters for trainer
+    num_epochs = num_epochs
+    num_gpu = num_gpu
+
+    # gpu_id = os.environ["CUDA_VISIBLE_DEVICES"]
+
+    # set up trainer
+    trainer = pl.Trainer(
+        max_epochs=num_epochs,
+        gpus=num_gpu,  # change to gpu_id
+        logger=TensorBoardLogger(save_dir=tune.get_trial_dir(), name="", version="."),
+        progress_bar_refresh_rate=0,
+        callbacks=[
+            TuneReportCallback(
+                {"loss": "val_loss"},
+                on="validation_end",
+            )
+        ],
+    )
+    trainer.fit(model, datamodule=dataset)
+
+
+@hydra.main(config_path="../config", config_name="config")
+# model=m2, dataset=m2, training=m2 in terminal
+def main(cfg: DictConfig, num_epochs=100, gpus_per_trial=1):
+
+    logger = logging.getLogger()
+
+    # restrict the number for cuda
+    os.environ["CUDA_VISIBLE_DEVICES"] = "1,4,3"
+
+    # define the parameter space
+    config = {
+        "enc_conv_c": tune.grid_search([5, 15, 20]),
+        "enc_kern": tune.grid_search([3]),
+        "enc_hidden": tune.grid_search([64, 128]),
+        "lr": tune.grid_search([1e-4, 1e-2]),
+        "weight_decay": tune.grid_search([1e-6, 1e-4]),
```

Comment:
is there a way you can change these grid search to continuous ranges like Jeff was saying? That would maek this a much more helpful example